Source/WebCore/css/calc/CSSCalcTree+Parser.cpp

```diff
@@ -628,61 +626,131 @@ static std::optional<TypedChild> consumeRound(CSSParserTokenRange& tokens, int d
     return std::nullopt;
 }
 
-static std::optional<Random::CachingOptions> consumeOptionalRandomCachingOptions(CSSParserTokenRange& tokens)
+static std::optional<Random::SharingFixed> consumeOptionalRandomSharingFixed(CSSParserTokenRange& tokens, ParserState& state)
+{
+    // <random-value-sharing-fixed> = fixed <number [0,1]>
+
+    ASSERT(tokens.peek().id() == CSSValueFixed);
+    tokens.consumeIncludingWhitespace();
+
+    // Use a non-property parsing state for the fixed number value to disconnect it from the current parse.
+    // FIXME: Add a mechanism to pass along the depth count when doing this so that we can limit stack usage.
+    auto numberParsingState = CSS::PropertyParserState { .context = state.propertyParserState.context };
+    auto number = CSSPropertyParserHelpers::MetaConsumer<CSS::Number<CSS::ClosedUnitRange>>::consume(tokens, numberParsingState);
+    if (!number)
+        return { };
+
+    return Random::SharingFixed {
+        .value = WTFMove(*number)
+    };
+}
+
+static Random::SharingOptions::Auto makeRandomSharingAuto(ParserState& state)
+{
+    return {
+        .property = state.propertyParserState.currentProperty,
+        .index = state.propertyParserState.cssRandomFunctionCount
+    };
+}
+
+static std::optional<Random::SharingOptions> consumeOptionalRandomSharingOptions(CSSParserTokenRange& tokens, ParserState& state)
 {
-    // <random-caching-options> = <dashed-ident> || per-element
+    // <random-value-sharing-options> = [ [ auto | <dashed-ident> ] || match-element ]
```

Comment:
I think the issue went with `element-shared`: https://github.com/w3c/csswg-drafts/issues/11742#issuecomment-2755039081

Reply:
We can change it if/when they update the spec.

Reply:
element-shared is the name kept by the resolution, I would go with that instead of match-element which was never resolved against. Maybe @fantasai can confirm